src/game.js

```diff
@@ -0,0 +1,99 @@
+"use strict"
+
+function Game() {
+  this._game = []
+  this.spareOrStrike = []
+  this.bonuses = []
+  this.runningTotal = 0
+}
+
+Game.prototype.addFrame = function(score){
+  this._game.push(score);
+};
+
+Game.prototype.returnLastElement = function(lastPosition, array){
+  return array[array.length - lastPosition];
+};
+
+Game.prototype.determineOutcomeofFrame = function(){
+  if(this.returnLastElement(1, this._game)[0] >= 10) {
+    this.spareOrStrike.push("strike");
+  } else if (this.returnLastElement(1, this._game)[0] + this.returnLastElement(1, this._game)[1] === 10) {
+      this.spareOrStrike.push("spare");
+    }
+    else {
+      this.spareOrStrike.push("neither");
+    }
+};
+
+Game.prototype.calculateBonuses = function(){
+  this.bonuses.push([])
+  this.calculateSpareBonus();
+  this.calculateStrikeBonus();
+  if (this.spareOrStrike[9] === "strike"){
+    var bonusFrame = new Frame()
+    bonusFrame.bowlFinalFrameBonuses(this, "strike")
+  } else if (this.spareOrStrike[9] === "spare"){
+    var bonusFrame = new Frame()
+    bonusFrame.bowlFinalFrameBonuses(this, "spare")
+  }
+};
+
+Game.prototype.addFinalFrameBonuses = function(bonusBowl){
+  for (var i = 0; i < bonusBowl.length; i++) {
+    this.bonuses[9].push(bonusBowl[i]);
+  }
+  if (this.spareOrStrike[8] === "strike"){
+    this.bonuses[8].push(bonusBowl[0])
+  }
+};
+
+Game.prototype.calculateSpareBonus = function(){
+  if (this.returnLastElement(2, this.spareOrStrike) === "spare") {
+    this.returnLastElement(2, this.bonuses).push(this.returnLastElement(1, this._game)[0]);
+  }
+};
+
+Game.prototype.calculateStrikeBonus = function(){
```

Comment:
This function is virtually unreadable!  Even if the logic is correct within the context of the design, it's incredibly difficult for a human to parse.